### DISCUSSION TITLE
High resolution simulations have run away heating at model top in L93

### DISCUSSION THREAD

**Discussion by JulioTBacmeister** [NCAR/amwg_dev#550] [Impact of vert res and model top]
Runs at ne120 and ne240 exhibit a slow runaway heating at the model top in L93.  After a year of simulation global mean model top T in an ne240 L93 run is nearly 2000K.  

The following line from rrtmg_state.F90 suggests rrtmg may shut itself off for pressures belwo 1Pa:
    num_rrtmg_levs = count( pref_edge(:) > 1._r8 ) ! pascals (1.e-2 mbar)

In L93 pref_edge[0] is ~0.4Pa. I wonder if this preventing radiation from balancing dynamical heating from momentum dissipation at the top.  

Runaway model-top heating has not shown up in ne30. I wonder if this could be because dissipation heating is small enough to be balanced by something else - diffusion? ...??

**Comment by brianpm** [comment]
2000K 🥵.

Your interpretation is correct, RRTMG (and also RRTMGP) should both be inactive at levels above 1Pa. I'm honestly not sure what happens up there... I think it should be equivalent to what WACCM does (?). Maybe @brian-eaton has an idea, or @cacraigucar knows if ALI-ARMS is supposed to be active.

**Comment by JulioTBacmeister** [comment]
Thanks Brian, maybe Francis Vitt should be added also (but I can't find his 'git handle' ☹️ ). Do you think it is possible to relax that upper boundary on RRTMG to say 0.1 Pa?  This is in the sponge anyway - so realism is optional. We just need cooling to kick in.

**Comment by fvitt** [comment]
WACCM has nonLTE longwave cooling acting at altitudes above 50 Pa (LW merge region between 50 and 5 Pa).  Try raising the upper boundary of RRTMG (as @JulioTBacmeister suggests) in this configuration where you are not using waccm physics.

**Comment by dan800** [comment]
Formichev isn't a very expensive routine if this doesn't work. I'd check to see if there's a heating term that comes out of the GW drag parameterization when parameterized waves are dissipated - might want to zero that at the UB.

**Comment by brian-eaton** [comment]
When I first tried running FMT with RRTMGP I ran into 2 issues at the upper boundary.  If RRTMGP gets pressures less than 1 Pa, or temperatures less than 160K it is not happy and returns an error.  The problem with pressure was an indexing bug that I fixed so we only pass layers with the top interface pressure less than 1 Pa to rrtmgp.  The temperature issue was "fixed" by a limiter in the code that moves CAM state data into the temperature array passed to rrtmgp.  The bottom line is that RRTMGP checks to be sure that input state values are within the ranges that its gas optics data are valid for.  I don't think that RRTMG has those same checks and it's happy to operate outside the valid range of its optics data... user beware.  I'm sure we've been bitten by this feature in the past...

**Comment by JulioTBacmeister** [comment]
Thanks all.  I will try raising the lid on RRTMG to 0.1Pa. Peter Lauritzen has also suggested we shut off dissipation heating in the top sponge, which I will do along with @dan800 's suggestion to turn off param GW dissipative heating

**Comment by JulioTBacmeister** [comment]
Here is the model top (monthly) in a one year ne240 simulation that has the runaway problem:
![image](https://github.com/NCAR/amwg_dev/assets/9221710/74ee058c-eb27-4135-97b5-7d62e113a489)

Do not have an equally long ne120 run, but looking at some shorter runs, I think removing dissipation heating and extending the top of RRTMG has eliminated the problem:
![image](https://github.com/NCAR/amwg_dev/assets/9221710/7cd4b426-61db-49b2-b4f3-147e886f5b1c)

**Reply by JulioTBacmeister** [reply]
@dan800 I checked the TTGW (total T tendency from GW) in the bad run (blue) above and it was nearly zero.   This is maybe not surprising since I've drastically scaled back the frontal GW at high-res. 

In any case, the orange run above still has dissipation heating from GW, but not from the dynamics.   But, I suspect all that really needs to be done is to allow RRTMG to work throughout the domain.  Do you think this is worth testing?

**Comment by JulioTBacmeister** [comment]
@brian-eaton and @brianpm, it appears that temperatures <160K occur rarely but consistently at the top of the L93 model. So, what should be done with RRTMGP? I'm not clear on what implications the limiter that @brian-eaton implemented might have.  Should we implement the Fomichev code from WACCM as @dan800 suggested?

**Comment by brian-eaton** [comment]
From a software point of view, the limiter is essentially implementing an extrapolation which says for temperatures below 160K use the optical properties valid at 160K.  If temperatures below 160K are completely unrealistic it would be best of course to add the missing physics so we don't have to deal with them.  If they are realistic we'd need to contact Robert Pincus about extending the validity of the gas optics.

I think dealing with trying to extend the top level of RRTMGP above 1 Pa will be more problematic as we can't use a limiter approach there.  Would need to contact Robert about that.
